packages/api-v2/src/hooks/p2p/entity/settings/p2p-settings/useSettings.ts

```diff
@@ -0,0 +1,107 @@
+import { useEffect } from 'react';
+import useSubscription from '../../../../../useSubscription';
+import { TSocketResponseData } from '../../../../../../types';
+import { useLocalStorage } from 'usehooks-ts';
+
+type TP2PSettings =
+    | (TSocketResponseData<'p2p_settings'>['p2p_settings'] & {
+          currency_list: {
+              display_name: string;
+              has_adverts: 0 | 1;
+              is_default?: 1;
+              text: string;
+              value: string;
+          }[];
+          float_rate_offset_limit_string: string;
+          is_cross_border_ads_enabled: boolean;
+          is_disabled: boolean;
+          is_payment_methods_enabled: boolean;
+          localCurrency?: string;
+          rate_type: 'float' | 'fixed';
+          reached_target_date: boolean;
+      })
+    | undefined;
+
+type TCurrencyListItem = {
+    display_name: string;
+    has_adverts: 0 | 1;
+    is_default?: 1;
+    text: string;
+    value: string;
+};
+
+const useSettings = () => {
```

Comment:
[Nit]: It would be beneficial to include a brief one-liner providing an explanation about this hook.

Reply:
honestly, I do not know the exact logic about it, as I just got /api-v2 in sync with /api, 
so whatever was added there, got added to /api-v2 

but added to my todo list and will comment it out with any next update,